### PR TITLE
perf(deps): replace `psl` with `tldts`

### PR DESCRIPTION
**Pull request by TonyRL**
It runs [fast and performant](https://remusao.github.io/posts/tldts-benchmarks.html).

<details><summary>Returned object is similar.</summary>

image
</details>

Background: https://github.com/salesforce/tough-cookie/pull/346

### CODE REVIEW

**Comment by DIYgod** [comment]
Thanks so much!

File: src/lib/rsshub.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ export function getPageRSSHub(data: {
 
     let parsedDomain;
     try {
-        parsedDomain = psl.parse(new URL(url).hostname);
+        parsedDomain = parse(new URL(url).hostname);
```
Comment (remusao) [inline src/lib/rsshub.ts:81]: Note: https://github.com/salesforce/tough-cookie/pull/346#discussion_r1429217096

Some of these might be relevant depending on your use-case. There are slight differences in default behavior between psl and tldts.

Comment (DIYgod) [reply src/lib/rsshub.ts:81]: Thank you so much, private domains will not be used in our extension, so there is no problem.

Comment (remusao) [reply src/lib/rsshub.ts:81]: If you want to shrink the bundle size a bit, there is a version of tldts that only ships with ICANN rules:  https://github.com/remusao/tldts/tree/master/packages/tldts-icann#tldts---blazing-fast-url-parsing-icann-rules-only

**Pull request merged**